### DISCUSSION TITLE
Linux build in userspace no sudo using conda

### DISCUSSION THREAD

**Discussion by ejhyer** [gnudatalanguage/gdl#1916] [Show and tell]
Hi GDL, 

I know the install scripts actually specifically check for installed conda and warn you about it, but I needed to make it go using conda, because I have no sudo privileges on the machines I need to use. I have seen how conda has interactions with non-conda configurations that can mess things up, but it does allow isolation of multiple independent conda configurations, which is very valuable in my workflows.

I built a conda environment with the following specification, and was able to successfully build GDL from source:
``` 
% conda create -n gdlenva -c conda-forge gsl gcc cxx-compiler cmake glpk libtirpc xorg-libx11 plplot wxwidgets geotiff netcdf4 h5py udunits2 omnia::eigen3 eccodes qhull-static graphicsmagick shapelib fftw freetype
% conda activate gdlenva
```
The only modification I needed to run GDL was to set PLPLOT_LIB as follows:
```
% find $CONDA_PREFIX -type d -name "*plplot*" | grep share # the path in /share is what you need
% export PLPLOT_LIB=<result of the operation above>
```
This was tested on a system with RHEL8 and another system using SLES. Test results are detailed in a comment downthread.

**Comment by ejhyer** [comment]
Test results using configuration above:
```
99% tests passed, 1 tests failed out of 214

Total Test time (real) = 216.72 sec

The following tests did not run:
  141 - test_mpi.pro (Skipped)

The following tests FAILED:
  112 - test_hdf5.pro (Failed)
```
The MPI I did not attempt-- this may be possible via conda, I didn't look into it.
The HDF fail results from `TEST_HDF5_OBJ_INFO` here is the relevant snippet:
```
; --- create a mock dataset (not implemented in GDL yet)
   if keyword_set(create) then begin
      a_number=123
      file_id = h5f_create(file_name)
      group_id = h5g_create(file_id, "a_group")
```
The error message is `% H5G_CREATE: Bad value` but the comment indicates that this test is not supposed to pass, I think?

**Comment by GillesDuvert** [comment]
Thanks @ejhyer , I leave @acoulais to take them into account, he is the 'founder' of the (historical now?) conda problem.
BTW, there should not be any dependency on PLPLOT_LIB those days, did you use 11.?
